binder/src/main/scala/PanamaCIRCTConverter.scala

```diff
@@ -635,8 +640,12 @@ class PanamaCIRCTConverter extends CIRCTConverter {
   def visitDefMemPort[T <: ChiselData](defMemPort: DefMemPort[T]): Unit = {
     val loc = util.convert(defMemPort.sourceInfo)
 
-    val op = util
-      .OpBuilder("chirrtl.memoryport", firCtx.currentBlock, loc)
+    val (parent, build) = firCtx.currentWhen match {
+      case Some(when) => (when.parent, (opBuilder: util.OpBuilder) => opBuilder.buildBefore(when.op))
```

Comment:
Does it recursively access to the outside of when block?

Reply:
FYI: https://github.com/llvm/circt/issues/1561

Reply:
Fixed. Thanks for the pointer.